src/components/ModelPages/widgets/FileInputAdapter.jsx

```diff
@@ -1,46 +1,47 @@
-import React, { useState } from 'react'
+import React from 'react'
 
 import DialogActions from 'cozy-ui/transpiled/react/DialogActions'
 import { ButtonLink } from 'cozy-ui/transpiled/react/Button'
 import Camera from 'cozy-ui/transpiled/react/Icons/Camera'
-import { useI18n } from 'cozy-ui/transpiled/react/I18n'
 import FileInput from 'cozy-ui/transpiled/react/FileInput'
 import { isMobileApp } from 'cozy-device-helper'
+import { useI18n } from 'cozy-ui/transpiled/react/I18n'
 
-import ID from 'assets/icons/Identity.svg'
-
-import { useStepperDialogContext } from '../Hooks/useStepperDialogContext'
-import CompositeHeader from '../CompositeHeader/CompositeHeader'
+import { useStepperDialogContext } from 'src/components/Hooks/useStepperDialogContext'
+import CompositeHeader from 'src/components/CompositeHeader/CompositeHeader'
 
-const Scan = ({ currentPage }) => {
+const FileInputAdapter = props => {
   const { t } = useI18n()
   const { nextPage } = useStepperDialogContext()
-  const { occurrence } = currentPage
-  const [currentOccurrence, setCurrentOccurrence] = useState(1)
 
   const onFileChange = file => {
     if (file) {
-      // Set CreatePaperProvider
-      if (currentOccurrence === occurrence) return nextPage()
+      props.onChange(file)
 
-      setCurrentOccurrence(prev => prev + 1)
+      // Timeout is required for "react-jsonschema-form" defines formData
+      setTimeout(() => {
+        nextPage()
+      }, 100)
```

Comment:
Do you have more info about this requirement? Is it possible that it takes more that 100ms?
Also 100ms is pretty long in terme of UI, aren't there any other mechanism we can use to avoid using an arbitrary setTimeout?

Reply:
Aujourd'hui, rjsf ne propose rien pour gérer mieux cette situation...
Mais un Timeout à 0 suffit, on va faire avec pour le moment.. :)

Reply:
Je confirme également que cela est temporaire, le dev qui suit va faire disparaitre cette nécessité 👍